src/server.h

```diff
@@ -385,6 +385,9 @@ extern int configOOMScoreAdjValuesDefaults[CONFIG_OOM_COUNT];
                                       memory eviction. */
 #define CLIENT_ALLOW_OOM (1ULL<<44) /* Client used by RM_Call is allowed to fully execute
                                        scripts even when in OOM */
+#define CLIENT_CUSTOM_AUTH (1ULL<<45) /* Indicates the client has module based auth in progress */
+#define CLIENT_CUSTOM_AUTH_RESULT (1ULL<<46) /* Indicates a Module has concluded custom auth with an
```

Comment:
If we take an earlier suggestion of just reprocessing the command repeatedly, I think this code also becomes unneeded.

Reply:
This code is used to check if the client was authenticated through custom authentication (via the RM_Authenticate* APis)

Reply:
I don't think you fully got my other comment, so AFAIK my comment is still true. In my suggestion there aren't auth results that are sticky outside of the call stack, as we immediately respond to the client.

Reply:
I replaced the `CLIENT_CUSTOM_AUTH` client flag with the `isClientModuleAuthInProgress` Macro here - https://github.com/redis/redis/pull/11659/commits/e0500097e976305e654fb1fac53e03251c9da288